src/lib.rs

```diff
@@ -14,26 +51,23 @@ use chrono::NaiveDateTime;
 /// ```rust
 /// use std::io::Write;
 /// use tempfile::NamedTempFile;
-/// use sagitta::{find_line};
+/// use sagitta::{find_line, read_file};
 ///
 /// let mut file = NamedTempFile::new().unwrap();
 /// writeln!(file, "test:one:big:thing:mem:1:20\ntest:two:job:thing:mem:2:100\ntest:three:job:thing:mem:3:234").unwrap();
 ///
 /// let line = "test:two:job:thing:mem:2:100";
 /// let file_path = String::from(file.path().to_str().unwrap());
+/// let open_file = read_file(file_path).unwrap();
 ///
-/// assert_eq!(Some(line.to_string()), find_line(file_path, 2).unwrap());
+/// assert_eq!(Some(line.to_string()), find_line(open_file, 2));
 /// ```
-pub fn find_line(f: String, id: i32) -> Result<Option<String>, anyhow::Error> {
-    let open_file = read_file(&f)?;
-
-    let line: Option<String> = open_file
+pub fn find_line<F: BufRead>(open_file: F, id: i32) -> Option<String> {
```

Comment:
This function returning an `Option` makes more sense, it will either find the line or not. So representing this as a `Option` over `Result` makes more sense.